### DISCUSSION TITLE
How to right-click a non item blank space to popup a menu

### DISCUSSION THREAD

**Discussion by AdminHcat** [ocornut/imgui#5372] [New Users Build/Link/Run issues ONLY!]
I want to right-click a non item  blank to popup a menu But I can only click on item anyway

![image](https://user-images.githubusercontent.com/50044729/171982791-016fe804-0e5f-4240-8deb-0686a3596277.png)

**Comment by PathogenDavid** [comment]
Thanks for the report! Next time please [open and issue and fill out the template](https://github.com/ocornut/imgui/issues/new). I know it's a lot, but seeing your code would've been helpful here. It may seem like you're just asking a question, but I think you actually found a bug. (We also only use discussions for building and font-related problems.)

Here is a workaround you can use until this is properly fixed:

```cpp
bool listBoxWasHovered = false;
if (ImGui::BeginListBox("Listboooox", ImVec2(0.f, 300.f)))
{
    ImGui::Selectable("Cherry");
    ImGui::Selectable("Mango");
    ImGui::Selectable("Pineapple");
    listBoxWasHovered = ImGui::IsWindowHovered(); // List boxes are technically child windows
    ImGui::EndListBox();
}

// This does about the same thing as ImGui::OpenPopupOnItemClick.
if (ImGui::IsMouseReleased(ImGuiMouseButton_Right) && (ImGui::IsItemHovered(ImGuiHoveredFlags_AllowWhenBlockedByPopup) || listBoxWasHovered))
{
    ImGui::OpenPopup("ListBoxContextMenu");
}

if (ImGui::BeginPopup("ListBoxContextMenu"))
{
    ImGui::Text("Hello, list box!");
    ImGui::EndPopup();
}
```

---------

I was able to repro this with the following: (Specifically @ 7b5a8e4f2bbc06eac3b913a7f7df67edff99c538)

```cpp
if (ImGui::BeginListBox("Listboooox", ImVec2(0.f, 300.f)))
{
    ImGui::Selectable("Cherry");
    ImGui::Selectable("Mango");
    ImGui::Selectable("Pineapple");
    ImGui::EndListBox();
}

ImGui::OpenPopupOnItemClick("ListBoxContextMenu", ImGuiPopupFlags_MouseButtonRight);

if (ImGui::BeginPopup("ListBoxContextMenu"))
{
    ImGui::Text("Hello, list box!");
    ImGui::EndPopup();
}
```

Specifically, the call to `IsItemHovered` in `OpenPopupOnItemClick` is failing when the mouse isn't over one of the items. The hover is being rejected right here:

https://github.com/ocornut/imgui/blob/7b5a8e4f2bbc06eac3b913a7f7df67edff99c538/imgui.cpp#L3539-L3546

This code was introduced to fix https://github.com/ocornut/imgui/issues/3851, and indeed things work as expected in b53b8f58dfbd4cb9f9b2a62e7a0a7e4500c491e9 and break with ee643b2ad919f9e85b973f3915d13a4fc0612a5a.

Not sure what the fix should be here, it seems like this is an area with a lot of subtle edge cases.
